src/pages/checkout/checkout.js

```diff
@@ -0,0 +1,355 @@
+import * as orderSummary from '../../components/order-summary/order-summary';
+import { activeElementMatches } from '../../js/utils';
+import * as Modal from '../../components/modal/modal';
+
+//let signInForm;
+let mainForm;
+let shippingSection;
+let paymentSection;
+let securityCodeTooltip;
+let securityCodeTooltipBtn;
+let securityCodeTooltipPopup;
+let securityCodeTooltipContent;
+let billingAddressCheckbox;
+let billingFormFields;
+let contactInfo;
+let shippingInfo;
+let paymentInfo;
+let editMode = false;
+
+const regex = {
+  'email': /^([A-Za-z0-9\-\/\:\;\(\)\$\&\"\=\,\?\*\#\%\^\+\_\.\|\[\]\{\}\<\>\\\'])+\@([A-Za-z0-9_\-])+\.([A-Za-z]{2,16})$/,
+  'postal-code': /^[A-Za-z]\d[A-Za-z][ -]?\d[A-Za-z]\d$/,
+  'phone-number': /^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$/im,
+  'credit-card-number': /^$[0-9]{14}/,
+  'expiry-date': /^[0-9]{2}[-\s\/\\]?[0-9]{2}$/,
+  'ccv': /^[0-9]{3,4}$/,
+};
+
+// START: Tooltip methods
+function testClick(event){
+  const clickInsideDropdown = event.target.matches('.payment-information__tooltip-popup, .payment-information__tooltip-popup *');
+  if(!clickInsideDropdown){
+    closeTooltip();
+  }
+}
+function testBlur(){
+  activeElementMatches('.payment-information__tooltip-popup *').then(function(focusInDropdown){
+    if(!focusInDropdown){
+      closeTooltip();
+    }
+  });
+}
+
+function openTooltip() {
+  securityCodeTooltip.classList.add('visible');
+  securityCodeTooltipPopup.setAttribute('aria-hidden', 'false');
+  securityCodeTooltipBtn.setAttribute('aria-expanded', 'true');
+
+  securityCodeTooltipContent.setAttribute('tabindex', 0);
+  securityCodeTooltipContent.focus();
+  securityCodeTooltipContent.addEventListener('blur', testBlur);
+  
+  const securityCodeTooltipClose = securityCodeTooltipPopup.querySelector('.payment-information__tooltip-popup__btn--close');
+  securityCodeTooltipClose.addEventListener('click', onClose);
+
+  document.addEventListener('click', testClick);
+}
+
+function onClose(event) {
+  event.stopPropagation();
+  closeTooltip();
+}
+
+function closeTooltip() {
+  securityCodeTooltip.classList.remove('visible');
+  securityCodeTooltipPopup.setAttribute('aria-hidden', 'true');
+  securityCodeTooltipContent.addEventListener('blur', testBlur).setAttribute('tabindex', -1);
+  securityCodeTooltipBtn.setAttribute('aria-expanded', 'false');
+  securityCodeTooltipBtn.focus();
+  document.removeEventListener('click', testClick);
+}
+
+function toggleTooltipIcon(event) {
+  event.stopPropagation();
+  if (securityCodeTooltip.classList.contains('visible')) {
+    closeTooltip();
+  } else {
+    openTooltip();
+  }
+}
+// END: Tooltip methods
```

Comment:
Again...this can wait until a future point, but I'd say moving this tooltip stuff into it's own component would be another good 'future work' candidate.

Reply:
Agreed. I was thinking the same thing at the time.